datasetinsights/datasets/transformers/coco.py

```diff
@@ -167,3 +168,182 @@ def _categories(self):
             categories.append(record)
 
         return categories
+
+
+class COCOKeypointsTransformer:
+    """Convert Synthetic dataset to COCO format.
+    This transformer convert Synthetic dataset into annotations
+    in person keypoint format
+    (e.g. person_keypoints_train2017.json, person_keypoints_val2017.json).
+    Note: We assume "valid images" in the COCO dataset must contain at least one
+    bounding box annotation. Therefore, all images that contain no bounding
+    boxes will be dropped. Instance segmentation are considered optional
+    in the converted dataset as some synthetic dataset might be generated
+    without it.
+    Args:
+        data_root (str): root directory of the dataset
+    """
+
+    # The annotation_definition.name is not a reliable way to know the type
+    # of annotation definition. This will be improved once the perception
+    # package introduced the annotation definition type in the future.
+    BBOX_NAME = r"^(?:2[dD]\s)?bounding\sbox$"
+    KPT_NAME = r"^(?:2[dD]\s)?keypoints$"
+    INSTANCE_SEGMENTATION_NAME = r"^instance\ssegmentation$"
+
+    def __init__(self, data_root):
+        self._data_root = Path(data_root)
+
+        ann_def = AnnotationDefinitions(
+            data_root, version=const.DEFAULT_PERCEPTION_VERSION
+        )
+        self._bbox_def = ann_def.find_by_name(self.BBOX_NAME)
+        self._kpt_def = ann_def.find_by_name(self.KPT_NAME)
+        try:
+            self._instance_segmentation_def = ann_def.find_by_name(
+                self.INSTANCE_SEGMENTATION_NAME
+            )
+        except NoRecordError as e:
+            logger.warning(
+                "Can't find instance segmentation annotations in the dataset. "
+                "The converted file will not contain instance segmentation."
+            )
+            logger.warning(e)
+            self._instance_segmentation_def = None
+
+        captures = Captures(
+            data_root=data_root, version=const.DEFAULT_PERCEPTION_VERSION
+        )
+        self._bbox_captures = captures.filter(self._bbox_def["id"])
+        self._kpt_captures = captures.filter(self._kpt_def["id"])
+        if self._instance_segmentation_def:
+            self._instance_segmentation_captures = captures.filter(
+                self._instance_segmentation_def
+            )
+
+    def execute(self, output):
+        """Execute COCO Transformer
+        Args:
+            output (str): the output directory where converted dataset will
+              be stored.
+        """
+        self._copy_images(output)
+        self._process_instances(output)
+
+    def _copy_images(self, output):
```

Comment:
This violates DRY principle which is exactly the same copy of code from `COCOInstancesTransformer`. I would recommend using inheritance/composition pattern to prevent code duplication.